sdk/core/azure-core/src/main/java/com/azure/core/util/polling/OperationResourcePollingStrategy.java

```diff
@@ -0,0 +1,265 @@
+     *
+     * @param httpPipeline an instance of {@link HttpPipeline} to send requests with
+     * @param context additional metadata to pass along with the request
+     */
+    public OperationResourcePollingStrategy(
+            HttpPipeline httpPipeline,
+            Context context) {
+        this.httpPipeline = httpPipeline;
+        this.context = context;
+    }
+
+
+    /**
+     * Gets the name of the operation location header. By default it's "Operation-Location".
+     * @return the name of the operation location header
+     */
+    public String getOperationLocationHeaderName() {
+        return OPERATION_LOCATION;
+    }
+
+    @Override
+    public Mono<Boolean> canPoll(Response<?> initialResponse) {
+        HttpHeader operationLocationHeader = initialResponse.getHeaders().get(getOperationLocationHeaderName());
+        return Mono.just(operationLocationHeader != null);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Mono<LongRunningOperationStatus> onInitialResponse(Response<?> response, PollingContext<T> pollingContext,
+                                                   TypeReference<T> pollResponseType) {
+        HttpHeader operationLocationHeader = response.getHeaders().get(getOperationLocationHeaderName());
+        HttpHeader locationHeader = response.getHeaders().get(LOCATION);
+        if (operationLocationHeader != null) {
+            pollingContext.setData(OPERATION_LOCATION, operationLocationHeader.getValue());
+        }
+        if (locationHeader != null) {
+            pollingContext.setData(LOCATION, locationHeader.getValue());
+        }
+        pollingContext.setData(HTTP_METHOD, response.getRequest().getHttpMethod().name());
+        pollingContext.setData(REQUEST_URL, response.getRequest().getUrl().toString());
+
+        if (response.getStatusCode() == 200
+                || response.getStatusCode() == 201
+                || response.getStatusCode() == 202
+                || response.getStatusCode() == 204) {
+            return Mono.just(LongRunningOperationStatus.IN_PROGRESS);
+        } else {
+            return Mono.error(new RuntimeException("Operation failed or cancelled: " + response.getStatusCode()));
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Mono<PollResponse<T>> poll(PollingContext<T> pollingContext, TypeReference<T> pollResponseType) {
+        HttpRequest request = new HttpRequest(HttpMethod.GET, pollingContext.getData(OPERATION_LOCATION));
+        Mono<HttpResponse> responseMono;
+        if (context == null) {
+            responseMono = httpPipeline.send(request);
+        } else {
+            responseMono = httpPipeline.send(request, context);
+        }
+        return responseMono.flatMap(res -> res.getBodyAsString()
+            .flatMap(body -> Mono.fromCallable(() ->
+                    serializer.<PollResult>deserialize(body, PollResult.class, SerializerEncoding.JSON))
```

Comment:
Should we make this configurable to support XML too?

Reply:
Added serializer config.